vm/src/vm/vm_memory/memory.rs

```diff
@@ -92,8 +92,8 @@ impl AddressSet {
 }
 
 pub struct Memory {
-    pub(crate) data: Vec<Vec<Option<MemoryCell>>>,
```

Comment:
Why did you made these 2 fields public? they are private to make it easier to refactor the representation and improve it. There are many methods for retrieving and inspecting the Memory and pretty-printing it (Display instance).

Reply:
Same, some macros access these fields directly so they must be public to use the macros out of tree. I'll try to think of another solution.